File: routes/GET/getAnimesInList.py
from classes import animesInList
from flask import make_response, jsonify

def get(profileId):
    try:
        animeInList = animesInList.AnimeInList(profileId=profileId)
        allAnimes = animeInList.getAllAnimes()
        match allAnimes["message"]:
            case "Animes found":
                resp = make_response(jsonify(allAnimes))
                resp.status_code=200
                return resp
            case "An error has occurred while getting the animes":
                resp = make_response(jsonify(allAnimes))
                resp.status_code=500
                return resp 
    except Exception as e:
        resp = make_response(jsonify({"message":"There was an error", "error":e.args}))
        resp.status_code=500
        return resp

File: src/routes/GET/getAnimesInList.ts
import AnimesInList from '../../classes/animesInList'
import type {Response, Request} from 'express'

export default async function getAnimesInList(req:Request, res:Response){
    try {
        const profileId =req.params.profileId
        const animes = await new AnimesInList("","","",profileId).getAll()
        switch(animes.message){
            case "success":
                res.statusCode = 200
                res.json(animes)
                break
            case "An error occurred":
                res.statusCode = 500
                res.json(animes)
            }

    } catch (error:any) {
        res.statusCode = 500
        res.json({message:"There was an error", error:error.message})
    }
}